src/what-season.js

```diff
@@ -1,6 +1,41 @@
 const CustomError = require("../extensions/custom-error");
 
-module.exports = function getSeason(/* date */) {
-  throw new CustomError('Not implemented');
-  // remove line with error and write your code here
+module.exports = function getSeason(userDate) {
+  if (typeof(userDate) === 'undefined' || userDate === null)
+  {
+    return 'Unable to determine the time of year!';
+  }
+  if (this.toString.call(userDate) === '[object Date]')
+  {
+    if (isNaN(userDate.getTime()))
+    {
+      throw CustomError(userDate + 'is not date format');
+    }
+  }
+  else
+    {
+      throw CustomError('Unable to determine the time of year!');
```

Comment:
You need to refactor this to not throw same error in different lines. line 6 & line 17 the same.